src/modules/storage/LedgerStorage.ts

```diff
@@ -119,4 +164,380 @@ export class LedgerStorage extends Storages
         });
     }
 
+    /**
+     * Put a transaction to database
+     * @param data a transaction data
+     * @param callback If provided, this function will be called when
+     * the database was finished successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    private putTransaction (block_height: number, tx: any, tx_index: number,
+        tx_hash: string, callback?: (err: Error | null) => void)
+    {
+        if (tx == undefined) {
+            if (callback != undefined)
```

Comment:
```suggestion
            if (callback != null)
```

Reply:
f no parameter is specified, it is `undefined`. Therefore, there is no need to modify it.

Reply:
So do I understand correctly we have three states:
- undefined
- null
- valid

And the undefined comes from the `?` ? In which case either `?` is redundant, or `| null` is, isn't it ?

Reply:
Ah nevermind, it's the parameter that can be `null`, not the callback itself